pkg/apis/internal.admin.acorn.io/v1/volumeclassresources_test.go

```diff
@@ -0,0 +1,289 @@
+package v1
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func TestVolumeClassResourcesAdd(t *testing.T) {
+	// Define test cases
+	testCases := []struct {
+		name     string
+		current  VolumeClassResources
+		incoming VolumeClassResources
+		expected VolumeClassResources
+	}{
+		{
+			name:     "add to empty VolumeClassResources resources",
+			current:  VolumeClassResources{},
+			incoming: VolumeClassResources{"foo": {resource.MustParse("1Mi")}},
+			expected: VolumeClassResources{"foo": {resource.MustParse("1Mi")}},
+		},
+		{
+			name:     "add to existing VolumeClassResources resources",
+			current:  VolumeClassResources{"foo": {resource.MustParse("1Mi")}},
+			incoming: VolumeClassResources{"foo": {resource.MustParse("1Mi")}},
+			expected: VolumeClassResources{"foo": {resource.MustParse("2Mi")}},
+		},
+		{
+			name:     "add where current has a resource specified with unlimited",
+			current:  VolumeClassResources{"foo": {UnlimitedQuantity()}},
+			incoming: VolumeClassResources{"foo": {resource.MustParse("1Mi")}},
+			expected: VolumeClassResources{"foo": {UnlimitedQuantity()}},
+		},
+		{
+			name:     "add where incoming has a resource specified with unlimited",
+			current:  VolumeClassResources{"foo": {resource.MustParse("1Mi")}},
+			incoming: VolumeClassResources{"foo": {UnlimitedQuantity()}},
+			expected: VolumeClassResources{"foo": {UnlimitedQuantity()}},
+		},
+		{
+			name:     "add where current and incoming have a resource specified with unlimited",
+			current:  VolumeClassResources{"foo": {UnlimitedQuantity()}},
+			incoming: VolumeClassResources{"foo": {UnlimitedQuantity()}},
+			expected: VolumeClassResources{"foo": {UnlimitedQuantity()}},
+		},
+		{
+			name: "add where current and incoming have a AllVolumeClasses specified with non-unlimited values",
+			current: VolumeClassResources{AllVolumeClasses: {
+				VolumeStorage: resource.MustParse("1Mi"),
+			}},
+			incoming: VolumeClassResources{AllVolumeClasses: {
+				VolumeStorage: resource.MustParse("1Mi"),
+			}},
+			expected: VolumeClassResources{AllVolumeClasses: {
+				VolumeStorage: resource.MustParse("2Mi"),
+			}},
+		},
+	}
+
+	// Run the test cases
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.current.Add(tc.incoming)
+			assert.True(t, tc.current.Equals(tc.expected))
+		})
+	}
+}
+
+func TestVolumeClassResourcesRemove(t *testing.T) {
```

Comment:
Can we add a test case here for the suspected issue?

Current has two volume classes, one is removed, and the other should still be there.